ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/PluginsReloadConfiguration.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods;
+
+import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.exception.InvalidJsonRpcParameters;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcErrorResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import org.hyperledger.besu.plugin.BesuPlugin;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class PluginsReloadConfiguration implements JsonRpcMethod {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private final Map<String, BesuPlugin> namedPlugins;
+
+  public PluginsReloadConfiguration(final Map<String, BesuPlugin> namedPlugins) {
+    this.namedPlugins = namedPlugins;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.PLUGINS_RELOAD_CONFIG.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
+    try {
+      final Optional<String> maybePluginName = requestContext.getOptionalParameter(0, String.class);
+      if (maybePluginName.isPresent() && namedPlugins.containsKey(maybePluginName.get())) {
+        reloadPluginConfig(maybePluginName.get(), namedPlugins.get(maybePluginName.get()));
+      } else {
+        LOG.info("Attempting to reload all plugins configurations.");
+        namedPlugins.forEach(this::reloadPluginConfig);
+      }
+      return new JsonRpcSuccessResponse(requestContext.getRequest().getId());
+    } catch (InvalidJsonRpcParameters invalidJsonRpcParameters) {
+      return new JsonRpcErrorResponse(
+          requestContext.getRequest().getId(), JsonRpcError.INVALID_PARAMS);
+    }
+  }
```

Comment:
Changing the default of an unspecified reload parameter to doing nothing or requiring a  parameter and throwing an error could be a potential solution to what @ajsutton was talking about

Reply:
Yeah. I will do the change

Reply:
Fixed https://github.com/hyperledger/besu/pull/261/commits/1cff3b25532f26868a9e29c998260cd862d63ac8